app/scripts/modules/core/navigation/states.provider.ts

```diff
@@ -0,0 +1,907 @@
+import {module} from 'angular';
+import {IState, IStateParamsService, IStateProvider, IUrlRouterProvider} from 'angular-ui-router';
+import {APPLICATION_READ_SERVICE, ApplicationReader} from 'core/application/service/application.read.service';
+
+require('./navigation.less');
+
+interface IMatch {
+  path: string;
+}
+
+interface IDeckState extends IState {
+  children?: IDeckState[];
+}
+
+interface IDeckStateProviderConfig {
+  buildApplication?: (mainView: string,
+                      relativeUrl?: string,
+                      baseChildren?: IDeckState[]) => IDeckState;
+  children?: Map<string, IDeckState>;
+}
+
+interface IDeckStateProvider {
+  state: (config: IDeckStateProviderConfig) => IDeckState;
+}
+
+interface IDeckStateParamService extends IStateParamsService {
+  account: string;
+  accountId: string;
+  application: string;
+  instanceId: string;
+  job: string;
+  name: string;
+  project: string;
+  provider: string;
+  region: string;
+  serverGroup: string;
+  vpcId: string;
+}
```

Comment:
I like the idea here and am going to tune it a bit, making a different interface for each state, which we'll be able to use where the `$stateParams` gets injected.